test/System.Web.Http.Test/Batch/DefaultHttpBatchHandlerTest.cs

```diff
@@ -281,6 +283,29 @@ public Task ParseBatchRequestsAsync_Throws_IfRequestIsNull()
             Assert.Equal("http://example.com/values", requests[1].RequestUri.AbsoluteUri);
         }
 
+        [Fact]
+        public async Task ParseBatchRequestsAsync_Returns_RequestsFromMultipartContent_WithUriSchemeSet_FromRequest()
+        {
+            DefaultHttpBatchHandler batchHandler = new DefaultHttpBatchHandler(new HttpServer());
+            HttpRequestMessage request = new HttpRequestMessage
+            {
+                Content = new MultipartContent("mixed")
+                {
+                    new HttpMessageContent(new HttpRequestMessage(HttpMethod.Get, "http://example.com/")),
+                    new HttpMessageContent(new HttpRequestMessage(HttpMethod.Post, "https://example.com/values"))
```

Comment:
Please add a part using a path-only relative URI e.g. `/api/values` since those are recommended and seem to be more common.

Reply:
I can't supply a relative Uri within a batch request. If I do so then ParseBatchRequestsAsync calls ReadAsMultipartAsync (an extension on HttpContent) and that will throw the following error.

Message: System.IO.IOException : Error reading MIME multipart body part.
---- System.InvalidOperationException : This operation is not supported for a relative URI.

Does this suggest that its absolute uri's by design or is this a different issue?

Reply:
Odd. But, if that's what enforced, so be it. On the off chance this is a bug I should open separately, please paste the full stack trace into this PR.